lib/spatial_features/importers/file.rb

```diff
@@ -0,0 +1,18 @@
+module SpatialFeatures
+  module Importers
+    class File < SimpleDelegator
+      def initialize(data, *args)
+        names = Unzip.names(data)
+        if names.any? {|name| name.ends_with? '.kml' }
+          __setobj__(KMLFile.new(data, *args))
+
+        elsif names.any? {|name| name.ends_with? '.shp' }
+          __setobj__(Shapefile.new(data, *args))
+
+        else
+          raise "Could not detect importer for file"
```

Comment:
We probably want to handle this error somewhere and add a validation error to the model.